### PR TITLE
STN-219: QA Feedback from Sprint 13

### PR DESCRIPTION
**Pull request by kaseybon**
# READY FOR REVIEW

## Summary
- Adjusts the `.hb-pill` styles when nested `divs` are used.
- Deletes the `.hb-subheading` utility.
- Removes default category styles from `.hb-card` and fixes the `hb-categories` utility when used on an `.entity-reference` field.
- Adds padding to `.hb-well`
- Fixes the mobile table headers for the table pattern. All rows should now have table headers.

## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. Confirm that the categories in horizontal card are unstyled and will then work when adding the `hb-categories` class.
1.  Add the `hb-categories` class to the research areas field on a person page and confirm the double border is gone.
1.  Add `hb-well` to any field on the person page and ensure that there is both padding and a background color.
1. Add the following `hb-pill-list` and `hb-pill-link-list` to a field that has multiple links, such as research areas, or info links.
1. View the courses page on mobile and check that all rows have their headings. Scroll all the way through the content, we didn't realize that this table was also broken but the breakage happened further down the page.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/js/tables/table-pattern.js [hunk 1]
```diff
@@ -4,19 +4,20 @@ let span = "div.hb-table-pattern__header > div.hb-table-pattern__row > span";
 let paragraph = "div.hb-table-pattern__header > div.hb-table-pattern__row > p";
 
 // retrieve table column headings
-let columnHeaders = document.querySelectorAll(`${div}, ${span}, ${paragraph}`);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/js/tables/table-pattern.js:7]: can you make the `lets` on line 1-3 into `consts` too?

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_fonts.scss [hunk 2]
```diff
@@ -22,10 +22,6 @@
   @include hb-heading-6;
 }
 
-.hb-subheading {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_fonts.scss:25]: 👍

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_general.scss [hunk 3]
```diff
@@ -64,6 +64,7 @@
     @include hb-global-color('background-color', 'gray-light');
   }
 
+  padding: hb-calculate-rems(20px) hb-calculate-rems(18px);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_general.scss:67]: do you think `hb-borderless` should get this padding too? I wasn't sure

**Review by silverli — CHANGES_REQUESTED** [review]
✅ categories look great on views or on fields

- Also, I noticed that when the hb-categories class is added to the horizontal card view wrapper, it changes the way this link looks on focus:
Screen Shot 2020-04-09 at 1 27 02 PM

- maybe we can just add this piece of information to the google doc about adding it as a style wrapper class to fields

✅ hb-well looks good with padding

⚠️ I couldnt get the hb-pills to look right. I added hb-pill-list, but this is what I see:
Screen Shot 2020-04-09 at 1 19 47 PM
Screen Shot 2020-04-09 at 1 19 52 PM

✅ the courses table looks good now

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_general.scss (hunk 3 above)
Comment (kaseybon) [reply docroot/themes/humsci/humsci_basic/src/scss/utilities/_general.scss:67]: @meranicosme would like for the content with `hb-borderless` to align with the rest of the content on the page. Only groups with a background or border should get padding.

**Comment by kaseybon** [comment]
Since Drupal doesn't give us any classes to target for the `hb-pill-list` I thought it was best to separate it into 2 utility classes: `hb-pill-list` and `hb-pill-link-list`. `hb-pill-list` should be used on lists that do not have links and `hb-pill-link-list` should be used when there are links. These classes have been documented in the Google Doc. I also added a mixin fix the icon color of external links in pills.

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_lists.scss [hunk 4]
```diff
@@ -11,33 +11,75 @@
...
+    // Normally this would be implemented as ul, ol { list-style-type: none; },
+    // display: none is used here to override the specificity of the
+    // ul & ol base element styles :not pseudo class selector
+    &::before {
+      display: none;
     }
   }
 
+  // for when the pill-list is a series of nested divs
+  div:not([class]) > div:not([class]) {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_lists.scss:58]: Thanks for making all these changes based on the different cases with labels. This work looks great 👍

**Review by silverli — APPROVED** [review]
Looks great! good work, thanks for sticking with my hb-pill requests

**Pull request merged**